admin_guide/sdn_troubleshooting.adoc

```diff
@@ -0,0 +1,547 @@
+in a pod that listens on a public IP address and "routes" that traffic
+inside the cluster.  The link:[OpenShift router] serves that purpose
+for HTTP, HTTPS (with SNI), WebSockets, or TLS (with SNI).
+
+Assuming you can't access an HTTP service from the outside of the
+cluster, let's start by reproducing the problem on the command line of
+the machine where things are failing.  Try:
+....
+curl -kv http://foo.example.com:8000/bar    # But replace the argument with your URL
+....
+
+If that works, are you reproducing the bug from the right place?  It
+is also possible that the service has some pods that work, and some
+that don't.  So jump ahead to the <<debugging-the-router>> section.
+
+
+If that failed, then let's resolve the DNS name to an IP address (assuming it isn't already one):
+....
+dig +short foo.example.com                  # But replace the hostname with yours
+....
+
+If that doesn't give back an IP address, it's time to troubleshoot
+DNS, but that's outside the scope of this guide.
+
+IMPORTANT: Make sure that the IP address that you got back is one that you expect to be running
+the router.  If it's not, fix your DNS.
+
+Next, use `ping -c _address_` and `tracepath _address_` to check that
+you can reach the router host.  It is possible that they will not
+respond to ICMP packets, in which case those tests will fail, but the
+router machine may be reachable.  In which case, try using the telnet
+command to access the port for the router directly:
+....
+telnet 1.2.3.4 8000
+....
+
+You may get:
+....
+Trying 1.2.3.4...
+Connected to 1.2.3.4.
+Escape character is '^]'.
+....
+
+If so, there's something listening on the port on the IP address.
+That's good.  Hit `ctrl-]` then hit the _enter_ key and then type
+`close` to quit telnet.  Move on to the <<debugging-the-router>>
+section to check other things on the router.
+
+Or you could get:
+....
+Trying 1.2.3.4...
+telnet: connect to address 1.2.3.4: Connection refused
+....
+
+Which tells us that the router is not listening on that port.  Please
+see the <<debugging-the-router>> section for more pointers on how to
+configure the router.
+
+
+Or if you see:
+[source,sh]
+....
+Trying 1.2.3.4...
+  telnet: connect to address 1.2.3.4: Connection timed out
+....
+
+Which tells us that you can't talk to anything on that IP address.
+Check your routing, firewalls, and that you have a router listening on
+that IP address.  To debug the router, see the
+<<debugging-the-router>> section. For IP routing and firewall issues,
+debugging that is beyond the purview of this guide.
+
+
+== Debugging the Router
+
+Now that you have an IP address, we need to `ssh` to that machine and
+check that the router software is running on that machine and
+configured correctly.  So let's `ssh` there and make sure the router
+is running:
+....
+# oc get endpoints --namespace=default --label=router
+NAMESPACE   NAME              ENDPOINTS
+default     router            10.1.0.4:80
+....
```

Comment:
Maybe obvious, but possibly mention running as a user w/ access to the "default" project.